package/MDAnalysis/coordinates/TPR.py

```diff
@@ -0,0 +1,80 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4
+
+from . import base
+from ..lib import util
+from .timestep import Timestep
+import MDAnalysis.topology.tpr.utils as tpr_utils
+import MDAnalysis.topology.tpr.setting as S
+
+
+import numpy as np
+
+
+class TPRReader(base.SingleFrameReaderBase):
+    # TODO: reduce duplication with `TPRparser`;
+    # we could share some state for the position
+    # in the binary file to avoid re-reading topology
+    # or perhaps combine the topology and coordinate reading
+    # with some inheritance shenanigans?
+    format = "TPR"
+    units = {"length": "nm", "velocity": "nm/ps"}
+    _Timestep = Timestep
+
+    def _read_first_frame(self):
+        # Read header/move over topology
+        # TODO: reduce duplication with TPRparser perhaps...
+        with util.openany(self.filename, mode="rb") as infile:
+            tprf = infile.read()
+        data = tpr_utils.TPXUnpacker(tprf)
+        try:
+            th = tpr_utils.read_tpxheader(data)  # tpxheader
+        except (EOFError, ValueError):
```

Comment:
Eventually a test that is triggered by wrong format would be good – I _assume_ we do this for the topology parser somewhere???